packages/ui/src/utils/tests/value-by-key.spec.ts

```diff
@@ -0,0 +1,56 @@
+import { describe, it, expect } from 'vitest'
+
+import { getValueByKey } from '../value-by-key'
+
+describe('value by key', () => {
+  const entries = [
+    {
+      props: {
+        option: 'string',
+        prop: 'prop',
+      },
+      expected: undefined,
+    },
+    {
+      props: {
+        option: { prop: 'string' },
+        prop: 'prop',
+      },
+      expected: 'string',
+    },
+    {
+      props: {
+        option: { prop: false },
+        prop: 'prop',
+      },
+      expected: false,
+    },
+    {
+      props: {
+        option: { prop: 'string' },
+        prop: undefined,
+      },
+      expected: { prop: 'string' },
+    },
+    {
+      props: {
+        option: { prop: 'string' },
+        prop: 1,
```

Comment:
It's initial behaviour:
- if something wrong with the type of the key (number, undef etc.) and helper isn't able to proceed it, option is returned itself;
- If the type of the key is correct, but target prop wasn't found in the provided object, it obviously returns undefined.

I was thinking about making both cases to return `undefined`, but it may create mega breaking change for nothing.